### DISCUSSION TITLE
Add custom page numbers to pdf

### DISCUSSION THREAD

**Discussion by Firestar-Reimu** [py-pdf/pypdf#2180] [Q&A]
I have a ebook which has page numbers of:

Cover, i, ii, iii, ... ix, x, 1, 2, 3, ... 514

But in pypdf I can only use 1, 2, 3, ... and it has a shift of 11 pages

How to add custom page numbers like "Cover" or latin alphabets or roman numbers?

**Comment by pubpub-zz** [comment]
what you are looking for is page_labels:
https://pypdf.readthedocs.io/en/stable/modules/PdfWriter.html?highlight=set_page_label#pypdf.PdfWriter.set_page_label

**Comment by pubpub-zz** [comment]
(to ease exchange I convert this into Q&A discussion)
